contracts/modules/ReserveAuction/V1/ReserveAuctionV1.sol

```diff
@@ -0,0 +1,528 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.10;
+
+import {ReentrancyGuard} from "@rari-capital/solmate/src/utils/ReentrancyGuard.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {ERC721TransferHelper} from "../../../transferHelpers/ERC721TransferHelper.sol";
+import {UniversalExchangeEventV1} from "../../../common/UniversalExchangeEvent/V1/UniversalExchangeEventV1.sol";
+import {OutgoingTransferSupportV1} from "../../../common/OutgoingTransferSupport/V1/OutgoingTransferSupportV1.sol";
+import {IncomingTransferSupportV1} from "../../../common/IncomingTransferSupport/V1/IncomingTransferSupportV1.sol";
+import {FeePayoutSupportV1} from "../../../common/FeePayoutSupport/FeePayoutSupportV1.sol";
+import {ModuleNamingSupportV1} from "../../../common/ModuleNamingSupport/ModuleNamingSupportV1.sol";
+
+/// @title Reserve Auction V1
+/// @author tbtstl <t@zora.co>
+/// @notice This contract allows users to list and bid on ERC-721 tokens with timed reserve auctions
+contract ReserveAuctionV1 is ReentrancyGuard, UniversalExchangeEventV1, IncomingTransferSupportV1, FeePayoutSupportV1, ModuleNamingSupportV1 {
+    /// @dev The indicator to pass all remaining gas when paying out royalties
+    uint256 private constant USE_ALL_GAS_FLAG = 0;
+    /// @notice The minimum amount of time left in an auction after a new bid is created
+    uint256 constant TIME_BUFFER = 15 minutes;
+    /// @notice The minimum percentage difference between the last bid amount and the current bid.
+    uint8 constant MIN_BID_INCREMENT_PERCENTAGE = 10;
+
+    /// @notice The ZORA ERC-721 Transfer Helper
+    ERC721TransferHelper public immutable erc721TransferHelper;
+
+    /// @notice The auction for a given NFT, if one exists
+    /// @dev ERC-721 token contract => ERC-721 token ID => Auction
+    mapping(address => mapping(uint256 => Auction)) public auctionForNFT;
+
+    /// @notice The metadata of an auction
+    /// @param seller The address of the seller creating the auction
+    /// @param currency The address of the ERC-20, or address(0) for ETH, required to bid
+    /// @param sellerFundsRecipient The address to send funds after the auction is settled
+    /// @param bidder The address of the highest bidder
+    /// @param finder The address of the referrer of the highest bid
+    /// @param findersFeeBps The fee to the referrer of the winning bid
+    /// @param amount The highest bid on the auction
+    /// @param duration The duration time of the auction after the first bid
+    /// @param startTime The start time of the auction
+    /// @param firstBidTime The time of the first bid
+    /// @param reservePrice The price to create the first bid
+    struct Auction {
+        address seller;
+        address currency;
+        address sellerFundsRecipient;
+        address bidder;
+        address finder;
+        uint16 findersFeeBps;
```

Comment:
One way to save 2 slots would be to limit the size of a few fields to a more reasonable number.

Duration can easily be `uint32` (probably not a great idea to have an auction beyond 100 years etc).

`uint32+uint16+address` is still less than `uint256` (one storage slot)

`firstBidTime` and `startTime` can both be `uint126` to save one slot as well

```
address
uint16 finderFeeBps
uint32 duration
uint128 firstBidTime
uint128 startTime
```

saves two slots over the current configuration.

Reply:
@iainnash hahaha i have these changes in the update. pushing soon!